homeassistant/components/isy994/sensor.py

```diff
@@ -32,42 +31,40 @@ class ISYSensorEntity(ISYNodeEntity):
     @property
     def raw_unit_of_measurement(self) -> str:
         """Get the raw unit of measurement for the ISY994 sensor device."""
-        if len(self._node.uom) == 1:
-            if self._node.uom[0] in UOM_FRIENDLY_NAME:
-                friendly_name = UOM_FRIENDLY_NAME.get(self._node.uom[0])
-                if friendly_name in (TEMP_CELSIUS, TEMP_FAHRENHEIT):
-                    friendly_name = self.hass.config.units.temperature_unit
-                return friendly_name
-            return self._node.uom[0]
-        return None
+        uom = self._node.uom
+
+        # Backwards compatibility for ISYv4 Firmware:
+        if isinstance(uom, list):
+            return UOM_FRIENDLY_NAME.get(uom[0], uom[0])
+        return UOM_FRIENDLY_NAME.get(uom)
 
     @property
     def state(self) -> str:
         """Get the state of the ISY994 sensor device."""
-        if self.is_unknown():
-            return None
-
-        if len(self._node.uom) == 1:
-            if self._node.uom[0] in UOM_TO_STATES:
-                states = UOM_TO_STATES.get(self._node.uom[0])
-                # TEMPORARY: Cast value to int until PyISYv2.
-                if int(self.value) in states:
-                    return states.get(int(self.value))
-            elif self._node.prec and self._node.prec != [0]:
-                str_val = str(self.value)
-                int_prec = int(self._node.prec)
-                decimal_part = str_val[-int_prec:]
-                whole_part = str_val[: len(str_val) - int_prec]
-                val = float(f"{whole_part}.{decimal_part}")
-                raw_units = self.raw_unit_of_measurement
-                if raw_units in (TEMP_CELSIUS, TEMP_FAHRENHEIT):
-                    val = self.hass.config.units.temperature(val, raw_units)
-
-                return str(val)
-            else:
-                return self.value
-
-        return None
+        if self.value == ISY_VALUE_UNKNOWN:
+            return STATE_UNKNOWN
+
+        uom = self._node.uom
+        # Backwards compatibility for ISYv4 Firmware:
+        if isinstance(uom, list):
+            uom = uom[0]
+        if not uom:
+            return STATE_UNKNOWN
+
+        states = UOM_TO_STATES.get(uom)
+        if states and states.get(self.value):
+            return states.get(self.value)
+        if self._node.prec and int(self._node.prec) != 0:
+            str_val = str(self.value)
+            int_prec = int(self._node.prec)
+            decimal_part = str_val[-int_prec:]
+            whole_part = str_val[: len(str_val) - int_prec]
```

Comment:
I realize this is existing, but it should be cleaned up later.

Reply:
Tagged in the HACS repo here with the proposed fix: https://github.com/shbatm/hacs-isy994/issues/61

Once tested properly, I will add this as an

There is also an as-of-yet-unreported existing issue where the temperatures are not converted to the proper units if they are whole numbers because of where the blocks are nested.